lib/jekyll/readers/data_reader.rb

```diff
@@ -37,10 +37,10 @@ def read_data_to(dir, data)
         path = @site.in_source_dir(dir, entry)
         next if @entry_filter.symlink?(path)
 
-        key = sanitize_filename(File.basename(entry, ".*"))
         if File.directory?(path)
-          read_data_to(path, data[key] = {})
+          read_data_to(path, data[entry] = {})
```

Comment:
I think we might still need to do `sanitize_filename` here to make this accessible in Liquid.